packages/gas-fee-controller/src/GasFeeControllerPolling.ts

```diff
@@ -0,0 +1,86 @@
+import type { RestrictedControllerMessenger } from '@metamask/base-controller';
+import { BaseControllerV2 } from '@metamask/base-controller';
+import { safelyExecute } from '@metamask/controller-utils';
+import type { NetworkClientId } from '@metamask/network-controller';
+import type { Json } from '@metamask/utils';
+import { v1 as random } from 'uuid';
+
+export default abstract class GasFeeControllerPolling<
+  N extends string,
+  S extends Record<string, Json>,
+  messenger extends RestrictedControllerMessenger<N, any, any, string, string>
+> extends BaseControllerV2<N, S, messenger> {
+  private readonly intervalLength = 1000;
+
+  private readonly networkClientIdTokensMap: Map<NetworkClientId, Set<string>> =
+    new Map();
+
+  private readonly intervalIds: Record<NetworkClientId, NodeJS.Timeout> = {};
+
+  start(networkClientId: NetworkClientId) {
+    const innerPollToken = random();
+    if (this.networkClientIdTokensMap.has(networkClientId)) {
+      //
+      const set = this.networkClientIdTokensMap.get(networkClientId);
+      set?.add(innerPollToken);
+    } else {
+      const set = new Set<string>();
+      set.add(innerPollToken);
+      this.networkClientIdTokensMap.set(networkClientId, set);
+    }
+    this.#poll(networkClientId);
+
+    // call _poll
+    // add the inner poll token to the poll tokens set
+    return innerPollToken;
+  }
+
+  stop({
+    pollingToken,
+    networkClientId,
+  }: {
+    pollingToken?: string;
+    networkClientId?: NetworkClientId;
+  } = {}) {
+    if (pollingToken && !networkClientId) {
+      throw new Error(
+        'networkClientId is required when pollingToken is passed',
```

Comment:
can we prepend the networkClientId to the polling token returned by `start()` or something to make it so that only pollingToken is required when calling this method?